usaspending_api/search/v2/views/spending_by_geography.py

```diff
@@ -381,40 +380,83 @@ def build_elasticsearch_search_with_aggregation(self, filter_query: ES_Q) -> Opt
         return search
 
     def build_elasticsearch_result(self, response: dict) -> Dict[str, dict]:
-        results = {}
+        def _key_to_geo_code(key):
+            return f"{code_to_state[key[:2]]['fips']}{key[2:]}" if (key and key[:2] in code_to_state) else None
+
+        # Get the codes
         geo_info_buckets = response.get("group_by_agg_key", {}).get("buckets", [])
+        geo_codes = [_key_to_geo_code(bucket["key"]) for bucket in geo_info_buckets if bucket.get("key")]
+
+        # Get the current geo info
+        current_geo_info = {}
+        if self.geo_layer == GeoLayer.STATE:
+            geo_info_query = (
+                PopCounty.objects.filter(state_code__in=geo_codes, county_number="000")
```

Comment:
These queries look pretty good, but double check that none of the data comes from CityCountyStateCode or StateData, sometimes the names or other data can differ between the tables, so just check which table transaction_search is joining to and make sure they line up.

Reply:
Yeah I was actually making my own `code_to_state` via StateData till I saw the import and we were using the one from `abbreviations.py` for the query django equivalent. I'm a little concerned that's just a dictionary that could be out of sync with the database but it's just states and their codes which rarely change.

As for the queries, I definitely use `StateData` in `spending_by_locations.py` for `self.location_type == LocationType.STATE_TERRITORY`, though it seems like it's used in the transaction search query? 
```
    LEFT OUTER JOIN (
        SELECT code, name, fips, MAX(id)
        FROM global_temp.state_data
        GROUP BY code, name, fips
    ) POP_STATE_LOOKUP ON (
        POP_STATE_LOOKUP.code = COALESCE(transaction_fpds.place_of_performance_state, transaction_fabs.place_of_perfor_state_code)
    )
    ...
    LEFT OUTER JOIN (
        SELECT code, name, fips, MAX(id)
        FROM global_temp.state_data
        GROUP BY code, name, fips
    ) RL_STATE_LOOKUP ON (
        RL_STATE_LOOKUP.code = COALESCE(transaction_fpds.legal_entity_state_code, transaction_fabs.legal_entity_state_code)
    )
```
Doesn't looks like I'm using `CityCountyStateCode` anywhere though.

Reply:
okay sounds good, I ran into an issue when I was testing where the county names varied slightly, but hopefully that was just for award search and the transactions won't have that issue.